core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -849,10 +849,11 @@ private DiscoveryNode findMaster() {
         List<ZenPing.PingResponse> pingResponses = new ArrayList<>();
         for (ZenPing.PingResponse pingResponse : fullPingResponses) {
             DiscoveryNode node = pingResponse.node();
-            if (masterElectionFilterClientNodes && (node.clientNode() || (!node.masterNode() && !node.dataNode()))) {
-                // filter out the client node, which is a client node, or also one that is not data and not master (effectively, client)
-            } else if (masterElectionFilterDataNodes && (!node.masterNode() && node.dataNode())) {
-                // filter out data node that is not also master
+            //nocommit we should rename this and its setting, also we ignore node.ingest, but maybe it's ok here
```

Comment:
I mean that we remove any mention of client nodes but we still have a setting called masterElectionFilterClientNodes. Not sure we should address this here though.